android/src/main/java/nordpol/TagDispatcherBuilder.java

```diff
@@ -0,0 +1,165 @@
+    private boolean enableSounds;
+    private boolean enableDispatchingOnUiThread;
+    private boolean enableBroadcomWorkaround;
+    private boolean enableReaderMode;
+    private boolean enableNdefCheck;
+
+    /**
+     * Constructor for TagDispatcherBuilder
+     * <p>
+     * By default the user will be prompted if NFC is unavailable. You can
+     * change this behavior with the method
+     * {@link #enableUnavailableNfcUserPrompt(boolean)}.
+     * <p>
+     * By default NFC interactions will produce sounds. You can change this
+     * behavior with the method {@link #enableSounds(boolean)}.
+     * <p>
+     * By default new tags will be dispatched on the UI thread. You can change
+     * this behavior with the method
+     * {@link #enableDispatchingOnUiThread(boolean)}.
+     * <p>
+     * By default TagDispatcher will use the Broadcom workaround. You can change
+     * this behavior with the method {@link #enableBroadcomWorkaround(boolean)}.
+     * <p>
+     * By default TagDispatcher will use ReaderMode. You can change this
+     * behavior with the method {@link #enableReaderMode(boolean)}.
+     * <p>
+     * By default TagDispatcher will check NFC NDEF. You can change this
+     * behavior with the method {@link #enableReaderMode(boolean)}.
+     * <p>
+     * Call {@link #build()} to build the TagDispatcher.
+     *
+     * @param activity               The Activity to attach the TagDispatcher to
+     * @param tagDiscoveredListener  A tagDiscoveredListener callback interface
+     */
+    public TagDispatcherBuilder (final Activity activity,
+                                 final OnDiscoveredTagListener tagDiscoveredListener) {
+        this.activity = activity;
+        this.tagDiscoveredListener = tagDiscoveredListener;
+        this.enableUnavailableNfcUserPrompt = true;
+        this.enableSounds = true;
+        this.enableDispatchingOnUiThread = true;
+        this.enableBroadcomWorkaround = true;
+        this.enableReaderMode = true;
+        this.enableNdefCheck = true;
+    }
+
+    /**
+     * Sets if TagDispatcher should handle the case of NFC not being available.
+     * <p>
+     * If NFC is disabled on the device and TagDispatcher is set as responsible
+     * then the user will receive a Toast asking them to enable NFC while
+     * being sent to the NFC settings. If the device completely lacks NFC and
+     * TagDispatcher is set as responsibe then the user will receive a Toast
+     * telling them that NFC isn't available on the device.
+     *
+     * @param enableUnavailableNfcUserPrompt  Should TagDispatcher handle NFC
+                                              unavailability?
+     * @return                                A new instance of
+                                              TagDispatcherBuilder
+     */
+    public TagDispatcherBuilder enableUnavailableNfcUserPrompt(boolean enableUnavailableNfcUserPrompt){
+        this.enableUnavailableNfcUserPrompt = enableUnavailableNfcUserPrompt;
```

Comment:
@Franzaine I think you should create  a new instance here instead so that we keep the class immutable. That would require a constructor that takes all properties. The same is valid for the rest of the methods.